openlibrary/plugins/upstream/account.py

```diff
@@ -387,19 +428,25 @@ def POST(self):
             expires = 3600 * 24 * 365 if remember.lower() == 'true' else ""
             web.setcookie('pd', int(audit.get('special_access')) or '', expires=expires)
             web.setcookie(config.login_cookie_name, web.ctx.conn.get_auth_token())
-            if audit.get('ia_email'):
-                ol_account = OpenLibraryAccount.get(email=audit['ia_email'])
-                if ol_account and ol_account.get_user().get_safe_mode() == 'yes':
-                    web.setcookie('sfw', 'yes', expires=expires)
-                if (
-                    ol_account
-                    and 'yrg_banner_pref' in ol_account.get_user().preferences()
-                ):
-                    web.setcookie(
-                        ol_account.get_user().preferences()['yrg_banner_pref'],
-                        '1',
-                        expires=(3600 * 24 * 365),
+            if audit.get('ia_email') and (
+                ol_account := OpenLibraryAccount.get(email=audit['ia_email'])
+            ):
+                _set_account_cookies(ol_account, expires)
+
+                if web.cookies().get("pda"):
+                    _notify_on_rpd_verification(
+                        ol_account, get_pd_org(web.cookies().get("pda"))
                     )
+                    _handle_pd_cookies(ol_account)
+                    add_flash_message(
+                        "info",
+                        _(
+                            "Thank you for registering an Open Library account and "
+                            "requesting special print disability access. You should receive "
+                            "an email detailing next steps in the process."
+                        ),
+                    )
```

Comment:
If we expect that this API is called via JSON, we likely don't need the flash message
```suggestion
```

Reply:
Good catch!  When authorized via Google Sign-In, patrons are redirected on the client-side:
https://github.com/internetarchive/openlibrary/blob/51b60377267b5cd6b47330b4bdbb9ebf5662fdfe/openlibrary/plugins/openlibrary/js/ia_thirdparty_logins.js#L25-L28

It's unclear to me how to best show such patrons a flash message.